From 4fdcc027fcc235805c7cc4bede6948b9a00afe1e Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Thu, 17 Oct 2024 10:44:19 +0100
Subject: [PATCH 6/6] libgcc, aarch64, Darwin: Fix ordering of function label
 and .cfi_startproc.

The newer tools apparently do not accept the existing order (which is likely
to cause quite some fallout with other OSS using asm).

Fixed here with a patch from Rich Townsend.  Resolves Issue #21.

Signed-off-by: Iain Sandoe <iains@gcc.gnu.org>
---
 libgcc/config/aarch64/lse.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git libgcc/config/aarch64/lse.S libgcc/config/aarch64/lse.S
index 87ee33bc52a5..c392ca27419c 100644
--- libgcc/config/aarch64/lse.S
+++ libgcc/config/aarch64/lse.S
@@ -197,9 +197,9 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 	.text
 	.balign	16
 	.private_extern	_\name
-	.cfi_startproc
 _\name:
 	BTI_C
+	.cfi_startproc
 .endm
 
 .macro	ENDFN name
-- 
2.47.0

